tmail-backend/webadmin/webadmin-mailbox/src/main/java/com/linagora/tmail/webadmin/cleanup/CleanupTrashService.java

```diff
@@ -64,20 +64,20 @@ public Mono<Task.Result> cleanupTrash() {
 
     private Mono<Task.Result> checkSettingAndCleanupTrashForSingleUser(Username username) {
         return Mono.from(jmapSettingsRepository.get(username))
-            .flatMap(jmapSettings -> getJMAPSettingsValue(jmapSettings, TRASH_CLEANUP_ENABLED_SETTING)
-                .map(trashCleanupEnabled -> checkSetting(trashCleanupEnabled, cleanupTrashForSingleUser(username,
-                    getJMAPSettingsValue(jmapSettings, TRASH_CLEANUP_PERIOD_SETTING).orElse(new JmapSettings.JmapSettingsValue(MONTHLY_PERIOD)).value())))
-                .orElse(Mono.just(Task.Result.COMPLETED)))
+            .flatMap(jmapSettings -> checkSettingAndCleanupTrashForSingleUser(username, jmapSettings))
             .onErrorResume(e -> {
                 LOGGER.error("Error while cleaning trash for user {}", username, e);
                 return Mono.just(Task.Result.PARTIAL);
             });
     }
 
-    private Mono<Task.Result> checkSetting(JmapSettings.JmapSettingsValue trashCleanupEnabled, Mono<Task.Result> mono) {
-        return Mono.just(trashCleanupEnabled).filter(jmapSettingsValue -> Boolean.parseBoolean(jmapSettingsValue.value()))
-            .flatMap(jmapSettingsValue -> mono)
-            .switchIfEmpty(Mono.just(Task.Result.COMPLETED));
+    private Mono<Task.Result> checkSettingAndCleanupTrashForSingleUser(Username username, JmapSettings jmapSettings) {
+        return getJMAPSettingsValue(jmapSettings, TRASH_CLEANUP_ENABLED_SETTING)
+            .map(jmapSettingsValue -> Mono.just(jmapSettingsValue).filter(trashCleanupEnabled -> Boolean.parseBoolean(trashCleanupEnabled.value()))
+                .flatMap(trashCleanupEnabled -> cleanupTrashForSingleUser(username,
+                    getJMAPSettingsValue(jmapSettings, TRASH_CLEANUP_PERIOD_SETTING).orElse(new JmapSettings.JmapSettingsValue(MONTHLY_PERIOD)).value()))
+                .switchIfEmpty(Mono.just(Task.Result.COMPLETED)))
+            .orElse(Mono.just(Task.Result.COMPLETED));
```

Comment:
Still not readable to me.
Please code needs to be instantly readable.
If a team member needs to think 10 minutes about what goes on then things needs to be re-written.

Eg:

` getJMAPSettingsValue(jmapSettings, TRASH_CLEANUP_PERIOD_SETTING).orElse(new JmapSettings.JmapSettingsValue(MONTHLY_PERIOD)).value()))`

 -> Sorry but this line is too long and complex. Likely we *could* have a constant for the default period?
 
 Maybe instead of writing `getJMAPSettingsValue(jmapSettings, TRASH_CLEANUP_PERIOD_SETTING)` we could write `jmapSettings.getValue(TRASH_CLEANUP_PERIOD_SETTING)`
 
This would give us

```
jmapSettings.getValue(TRASH_CLEANUP_PERIOD_SETTING).orElse(DEFAULT_PERIOD)
```

Also we are mixing two monad types which makes the reasonning extra hard. (are we mapping the mono or the optional?) Maybe turning the Optional into a mono would ease things up? Or Maybe it is easier to have a boolean upfront?

EG

```
boolean trashCleanupEnabled = jmapSettings.getValue(TRASH_CLEANUP_ENABLED_SETTING).orElse(false);
Duration trashCleanupPeriod = jmapSettings.getValue(TRASH_CLEANUP_PERIOD_SETTING).orElse(MONTHLY_PERIOD);

if (trashCleanupEnabled) {
    return cleanupTrashForSingleUser(username, trashCleanupPeriod); // Make it return a Result
}
return Mono.just(COMPLETED);
```

Easier to read, right?

We spend 90% of our time reading code, so making extra cognitive efforts to write good code is worth it.